pennylane/ops/qutrit/non_parametric_ops.py

```diff
@@ -202,3 +203,200 @@ def adjoint(self):
         op = TClock(wires=self.wires)
         op.inverse = not self.inverse
         return op
+
+
+class TAdd(Operation):
+    r"""TAdd(wires)
+    The 2-qutrit controlled add gate
+
+    The construction of this operator is based on definition 7 from
+    `Yeh et al. (2022) <https://arxiv.org/abs/2204.00552>`_.
```

Comment:
Even though the reference is provided, I think it would be helpful to show the action explicitly because it is hard to glean this from a 9 x 9 matrix. We'll just have to check that it renders correctly.

```suggestion
    `Yeh et al. (2022) <https://arxiv.org/abs/2204.00552>`_. 
    It performs the controlled :class:`~.TShift` operation, and sends
    :math:`\hbox{TAdd} \vert i \rangle \vert j \rangle = \vert i \rangle \vert i + j \rangle`, 
    where addition is taken modulo 3. The matrix representation is
```